libafl/src/executors/inprocess/mod.rs

```diff
@@ -432,55 +436,65 @@ pub fn run_observers_and_save_state<E, EM, OF, Z>(
     input: &<E::State as UsesInput>::Input,
     fuzzer: &mut Z,
     event_mgr: &mut EM,
-    exitkind: ExitKind,
+    exit_kind: ExitKind,
 ) where
     E: HasObservers,
     EM: EventFirer<State = E::State> + EventRestarter<State = E::State>,
     OF: Feedback<E::State>,
     E::State: HasExecutions + HasSolutions + HasCorpus,
-    Z: HasObjective<Objective = OF, State = E::State>,
+    Z: HasObjective<Objective = OF, State = E::State>
+        + HasScheduler<State = E::State>
+        + ExecutionProcessor,
 {
-    let mut observers = executor.observers_mut();
-
-    observers
-        .post_exec_all(state, input, &exitkind)
-        .expect("Observers post_exec_all failed");
-
-    let interesting = fuzzer
-        .objective_mut()
-        .is_interesting(state, event_mgr, input, &*observers, &exitkind)
-        .expect("In run_observers_and_save_state objective failure.");
-
-    if interesting {
-        let executions = *state.executions();
-        let mut new_testcase = Testcase::with_executions(input.clone(), executions);
-        new_testcase.add_metadata(exitkind);
-        new_testcase.set_parent_id_optional(*state.corpus().current());
-
-        if let Ok(mut tc) = state.current_testcase_mut() {
-            tc.found_objective();
+    let observers = executor.observers_mut();
+    let scheduler = fuzzer.scheduler_mut();
+
+    let _ = scheduler.on_evaluation(state, input, &*observers);
+
+    let res = fuzzer.check_results(state, event_mgr, input, &*observers, &exit_kind);
+    if let Ok(exec_res) = res {
+        let _ = fuzzer.process_execution(state, event_mgr, input, &exec_res, &*observers);
+
+        // Now send off the event
+        // unlike in fuzzer/mod.rs we never serialize here.
+        // (because it doesn't work with generics)
+        match exec_res {
+            ExecuteInputResult::Corpus => {
+                if event_mgr.should_send() {
+                    // TODO set None for fast targets
+                    let _ = event_mgr.fire(
+                        state,
+                        Event::NewTestcase {
+                            input: input.clone(),
+                            observers_buf: None,
+                            exit_kind,
+                            corpus_size: state.corpus().count(),
+                            client_config: event_mgr.configuration(),
+                            time: current_time(),
+                            executions: *state.executions(),
+                            forward_id: None,
+                            #[cfg(all(unix, feature = "std", feature = "multi_machine"))]
+                            node_id: None,
+                        },
+                    );
+                }
+            }
+            ExecuteInputResult::Solution => {
+                let executions = *state.executions();
+                let _ = event_mgr.fire(
```

Comment:
*cough*

Reply:
Oh wow for `fire` we ignore the result very often. That's gonna bite us at some point..